# Download and Cache Mistral 7B model
CACHE_DIR is set to /nvmefs1/test_user/cache/, make sure to change this to the directory of choice

In [2]:
import subprocess
import sys
import importlib
from transformers import AutoModelForCausalLM, AutoTokenizer
import torch


print(f"Transformers version: {transformers.__version__}")

Transformers version: 4.36.0.dev0


## Restart kernel to ensure transformimport transformersers 4.36 is loaded

In [5]:
import transformers
from transformers import AutoModelForCausalLM, AutoTokenizer
from transformers import AutoTokenizer, AutoModelForCausalLM, BitsAndBytesConfig
import torch
import os
print(f"Transformers version: {transformers.__version__}")

Transformers version: 4.36.0.dev0


In [7]:
CACHE_DIR='/nvmefs1/test_user/cache/'
model_cache_dir=os.path.join(CACHE_DIR,'model_cache/mistral_7b_model_tokenizer')
print(model_cache_dir)
tokenizer_cache_dir=os.path.join(CACHE_DIR,'model_cache/mistral_7b_model_tokenizer')
print(tokenizer_cache_dir)

/nvmefs1/test_user/cache/model_cache/mistral_7b_model_tokenizer
/nvmefs1/test_user/cache/model_cache/mistral_7b_model_tokenizer


In [8]:
model_id = "mistralai/Mistral-7B-Instruct-v0.1"
tokenizer = AutoTokenizer.from_pretrained(model_id,cache_dir=tokenizer_cache_dir)
model = AutoModelForCausalLM.from_pretrained(model_id,cache_dir=model_cache_dir)



Loading checkpoint shards:   0%|          | 0/2 [00:00<?, ?it/s]

In [9]:
print("Saiving model to... ", os.path.join(CACHE_DIR,'model/mistral_7b_model_tokenizer'))
model.save_pretrained(os.path.join(CACHE_DIR,'model/mistral_7b_model_tokenizer'))
print("Saiving model to... ", os.path.join(CACHE_DIR,'model/mistral_7b_model_tokenizer'))
tokenizer.save_pretrained(os.path.join(CACHE_DIR,'model/mistral_7b_model_tokenizer'))
      

Saiving model to...  /nvmefs1/test_user/cache/model/mistral_7b_model_tokenizer
Saiving model to...  /nvmefs1/test_user/cache/model/mistral_7b_model_tokenizer


('/nvmefs1/test_user/cache/model/mistral_7b_model_tokenizer/tokenizer_config.json',
 '/nvmefs1/test_user/cache/model/mistral_7b_model_tokenizer/special_tokens_map.json',
 '/nvmefs1/test_user/cache/model/mistral_7b_model_tokenizer/tokenizer.model',
 '/nvmefs1/test_user/cache/model/mistral_7b_model_tokenizer/added_tokens.json',
 '/nvmefs1/test_user/cache/model/mistral_7b_model_tokenizer/tokenizer.json')

# Check if Model successfully downloaded by loading the model from our local directory

In [10]:
print("Loading tokenizer from... ", os.path.join(CACHE_DIR,'model/mistral_7b_model_tokenizer'))

tokenizer = AutoTokenizer.from_pretrained( os.path.join(CACHE_DIR,'model/mistral_7b_model_tokenizer'),cache_dir=tokenizer_cache_dir)
print("Loading model from... ", os.path.join(CACHE_DIR,'model/mistral_7b_model_tokenizer'))
model = AutoModelForCausalLM.from_pretrained(os.path.join(CACHE_DIR,'model/mistral_7b_model_tokenizer'),
                                             cache_dir=model_cache_dir,
                                            torch_dtype=torch.bfloat16,
                                            device_map = "auto",
)

Loading tokenizer from...  /nvmefs1/test_user/cache/model/mistral_7b_model_tokenizer
Loading model from...  /nvmefs1/test_user/cache/model/mistral_7b_model_tokenizer


Loading checkpoint shards:   0%|          | 0/6 [00:00<?, ?it/s]